### PR TITLE
delegated operators

### PR DESCRIPTION
**Pull request by nebulae**
Implementation of delegated operators!

## CLI examples

Launch a local service that will run all delegated operations:

```shell
fiftyone delegated launch
```

List delegated operations:

```shell
# List all operations
fiftyone delegated list

# List specific operations
fiftyone delegated list \
    --dataset quickstart \
    --operator @voxel51/io/export_samples \
    --state COMPLETED \
    --sort-by COMPLETED_AT \
    --limit 2
```

Cleanup delegated operations that you no longer want to keep records for:

```shell
# Delete all failed operations associated with a given dataset
fiftyone delegated cleanup --dataset quickstart --state FAILED

# Delete all delegated operations associated with non-existent datasets
fiftyone delegated cleanup --orphan

# Print information about operations rather than actually deleting them
fiftyone delegated cleanup --orphan --dry-run        
```

## Programmatically run an operation

```py
import fiftyone as fo
import fiftyone.zoo as foz
import fiftyone.operators as foo

dataset = foz.load_zoo_dataset("quickstart")

ctx = dict(dataset=dataset)

params = dict(
    target="DATASET",
    export_dir="/tmp/coco",
    export_type="LABELS_ONLY",
    dataset_type="COCO",
    label_field="ground_truth",
)

foo.execute_operator("@voxel51/io/export_samples", ctx, params)
```

### CODE REVIEW

File: fiftyone/operators/executor.py [hunk 1]
```diff
@@ -202,25 +295,11 @@ def selected(self):
         """The list of selected sample IDs or an empty list."""
         return self.request_params.get("selected", [])
 
-    @property
```
Comment (brimoor) [inline fiftyone/operators/executor.py:205]: Inadvertently deleted? `ctx.selected_labels` was a feature that @ritch added recently. Maybe its a release branch vs develop issue?

File: fiftyone/operators/executor.py [hunk 2]
```diff
@@ -78,41 +86,126 @@ def to_json(self):
         }
 
 
-@coroutine_timeout(seconds=fo.config.operator_timeout)
```
Comment (brimoor) [inline fiftyone/operators/executor.py:81]: Don't we still need timeout? (now on `execute_or_delegate_operator()`)

File: fiftyone/operators/delegated.py [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+    def queue_operation(
+        self,
+        operator: str,
+        delegation_target: str = None,
+        context: ExecutionContext = None,
+    ) -> DelegatedOperationDocument:
+        """Returns a queued :class:`fiftyone.core.odm.DelegatedOperationDocument` instance
+        for the operation.
+        Returns:
```
Comment (brimoor) [inline fiftyone/operators/delegated.py:47]: Looks like copy + pasting of code removed some blank lines in your docstrings. Sphinx needs the whitespace in order to correctly render the docs (applies throughout PR)

**Review by ritch — APPROVED** [review]

**Review by brimoor — APPROVED** [review]
LGTM 🚀

**Pull request merged**